specification.md

```diff
@@ -377,8 +377,7 @@ Referenced resource must conform to the specifications [Workflow Events JSON Sch
 | Parameter | Description | Type | Required |
 | --- | --- | --- | --- |
 | name | Unique function name | string | yes |
-| resource | Function resource (URI) | string | yes |
-| type | Function type | string | no |
+| operation | Combination of the function/service OpenAPI definition URI and the operationID of the operation that needs to be invoked, separated by a '#'. For example 'https://petstore.swagger.io/v2/swagger.json#getPetById' | string | no |
```

Comment:
i think its not required so that users that just want to use metadata do not have to define it. if it makes more sense to enforce it (require) thats fine. 

What about the "type" param. what did we end up deciding in meeting how to proceed ? Add it back in and make it default to "openapi" or something? I cant remember :)

Reply:
Yeah, the type would make more sense. So the type + resource fields would be required. If type is not informed, we would assume `openapi` and the resource a link to the meta information?

Reply:
I was under the impression that this was the conclusion (add 'type' back, but default it to 'openapi'), but I don't think it was clearly stated in the meeting that this was the conclusion.  I'm personally in support of this, as I think it brings in first-class support for openapi and encourages it's use but doesn't block use-cases for invoking other types of 'functions' that aren't natively RESTful.

On a related note, is this field extensible to support other API/RPC spec types in the future?  (Not specifically looking for WSDL/SOAP support, but perhaps consider that as an example).

Reply:
ok will add "type" back and put some documentation around it, specifically that runtime implementors need to know that it is a completely not-portable parameter. 
as far as extensibility I think can deal with http(restful) service now as that includes the majority of cases and if need to incorporate other protocols find a good standard/specification that we can use

Reply:
> as far as extensibility I think can deal with http(restful) service now as that includes the majority of cases and if need to incorporate other protocols find a good standard/specification that we can use . wdyt?

Totally agreed on not adding any other protocol/spec right now.  

It's probably best not to over-think this 'someday maybe we'll add another protocol/spec' hypothetical (so feel free to ignore the question), just curious your thoughts on this since you've no doubt spent more time looking at this than I 
 have. =)

If some fancy new protocol/spec comes along in the future though, does the single attribute named 'operation' make sense to re-use for that?   Taking WSDL as an example, can a method described there be similarly identified by a single URI like OAS operation, and will it make sense to refer to it as 'operation'?

Reply:
I thought of it similarly to what we are doing with event definitions and the CloudEvents specification. There we use parameters named "source" and "type" which directly map to equally named CE context attributes. 

Since we use the OpenAPI "operationId" as the unique identifier of the operation of the service to be invoked I thought this would fit with what we are trying to do, but if a different parameter name would be more suited we can definitely change to it. What would you propose?

Reply:
Not knowing what potential future spec/standard we might allow for in the future, I think it's very reasonable to go with 'operation' since it makes a clear connection to OpenAPI concepts, which is what the spec supports now.  Also, I don't think it's a stretch to use 'operation' parameter to reference some other type of spec that might come along, even if the language for that type of spec calls it a 'method' or a 'function', instead of an 'operation'.

On the other hand, I don't think it's unreasonable to keep the 'resource' attribute name, but mandate that when the 'type=openapi' that the resource URI must be a valid URI pointing to an operation in an accessible open-api spec file (either network or locally accessible to the runtime).  I think this comes at the cost of extra ambiguity, but it does seem to leave things more open on the types of 'function' that this workflow spec might support.

I worry that I'm overthinking this, bike-shedding on a name here, so know that I think 'operation' is a fine choice.  =)